### DISCUSSION TITLE
How to get header value in post-request script from response object

### DISCUSSION THREAD

**Discussion by chiflux** [hoppscotch/hoppscotch#2666] [Q&A]
How can I get a specific header value from the response object so that I can set it as an env variable?
Currently, I only get "Could not execute post-request script" but no detailed error message.

Example:
pw.env.set("token", pw.response.headers.get('token'));

**Comment by MikeKrueger75** [comment]
Hi,
it doesn't work, because `headers` is an object without get-method. 
Here is my simple solution for this problem:
```javascript
function findAndSetLocation(item){
  if(item['key']=='token'){
    pw.env.set("token", item['value'])
  }
}

var headers = pw.response.headers
headers.forEach(findAndSetLocation)
```
I hope this is helpful to you. 
PS: there will be more efficient solutions

**Comment by kszinhu** [comment]
Hey there, yeah, `headers` is a `TypedArray` (without a get-method). I prefer converting `response.headers` into a `Map` when setting multiple variables in bulk:

```js
const headers = new Map(pw.response.headers.map(item => [item['key'], item['value']]))

pw.env.set("uid", headers.get("uid"))
pw.env.set("expire", headers.get("expiry"))
pw.env.set("client", headers.get("client"))
pw.env.set("token", headers.get("access-token"))
pw.env.set("authorization", headers.get("authorization"))
```
